packages/web/app/src/components/layouts/target-selector.tsx

```diff
@@ -52,65 +74,73 @@ export function TargetSelector(props: {
-      {targetEdges?.length && currentOrganization && currentProject && currentTarget ? (
-        <>
-          <Select
-            value={props.currentTargetSlug}
-            onValueChange={id => {
-              void router.navigate({
-                to: '/$organizationSlug/$projectSlug/$targetSlug',
-                params: {
-                  organizationSlug: props.currentOrganizationSlug,
-                  projectSlug: props.currentProjectSlug,
-                  targetSlug: id,
-                },
-              });
+      {showOrganization ? (
+        currentOrganization ? (
+          <Link
+            to="/$organizationSlug"
+            params={{
+              organizationSlug: currentOrganization.slug,
             }}
+            className="max-w-[200px] shrink-0 truncate font-medium"
           >
+            {currentOrganization.slug}
+          </Link>
+        ) : (
+          <div className="h-5 w-48 max-w-[200px] animate-pulse rounded-full bg-gray-800" />
+        )
+      ) : null}
+      {showOrganization ? <div className="italic text-gray-500">/</div> : <></>}
+      {showProject ? (
+        currentOrganization && currentProject ? (
+          <Link
+            to="/$organizationSlug/$projectSlug"
+            params={{
+              organizationSlug: props.currentOrganizationSlug,
+              projectSlug: props.currentProjectSlug,
+            }}
+            className="max-w-[200px] shrink-0 truncate font-medium"
+          >
+            {currentProject.slug}
+          </Link>
+        ) : (
+          <div className="h-5 w-48 max-w-[200px] animate-pulse rounded-full bg-gray-800" />
+        )
+      ) : null}
+      {showProject ? <div className="italic text-gray-500">/</div> : null}
+      {(targetEdges?.length && currentOrganization && currentProject && currentTarget) ||
+      isOptional ? (
+        <>
+          <Select value={props.currentTargetSlug} onValueChange={onValueChangeFunc}>
             <SelectTrigger variant="default" data-cy="target-picker-trigger">
               <div className="font-medium" data-cy="target-picker-current">
-                {currentTarget.slug}
+                {isOptional ? (
+                  <SelectValue placeholder="Pick an option" />
+                ) : (
+                  (currentTarget?.slug ?? '')
+                )}
               </div>
             </SelectTrigger>
             <SelectContent>
-              {targetEdges.map(edge => (
-                <SelectItem
-                  key={edge.node.slug}
-                  value={edge.node.slug}
-                  data-cy={`target-picker-option-${edge.node.slug}`}
-                >
-                  {edge.node.slug}
+              {isOptional ? (
+                <SelectItem key="empty" value="empty" data-cy="project-picker-option-Unassigned">
+                  Unassigned
                 </SelectItem>
-              ))}
+              ) : (
+                <></>
+              )}
```

Comment:
This can be `isOptional  && ()`, to avoid the ternary. TBH I don't care too much about using a ternary here, but rather having the else condition being a Fragment instead of simply null.